singlevenue/src/main/java/com/mappedin/examples/singlevenue/SearchCallback.java

```diff
@@ -0,0 +1,219 @@
+package com.mappedin.examples.singlevenue;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.view.Gravity;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.BaseAdapter;
+import android.widget.EditText;
+import android.widget.GridView;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import com.mappedin.sdk.Location;
+import com.mappedin.sdk.SearchDelegate;
+import com.mappedin.sdk.SearchResult;
+import com.mappedin.sdk.SearchResultType;
+import com.mappedin.sdk.SmartSearch;
+
+import static com.mappedin.examples.singlevenue.MainActivity.PICK_CONTACT_REQUEST;
+
+/**
+ * implement SearchDelegate, handle results for search and suggest
+ */
+class SearchCallback implements SearchDelegate {
+
+    private Activity activity;
+    private SmartSearch smartSearch;
+    private EditText searchBar;
+    private ListView suggestListView;
+    private GridView searchGridView;
+
+    public SearchCallback(Activity activity, SmartSearch smartSearch, EditText searchBar, ListView suggestListView, GridView searchGridView) {
+        this.activity = activity;
+        this.smartSearch = smartSearch;
+        this.searchBar = searchBar;
+        this.suggestListView = suggestListView;
+        this.searchGridView = searchGridView;
+    }
+
+    @Override
+    public void suggestion(String quary, final String[] suggests) {
+        if (suggests.length > 1 || (suggests.length == 1 && !suggests[0].equals(quary))) {
+            SuggestListAdapter adapter =
+                    new SuggestListAdapter(activity, suggests);
+            suggestListView.setAdapter(adapter);
+            suggestListView.setVisibility(View.VISIBLE);
+            suggestListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+                @Override
+                public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                    String selected = suggests[position];
+                    searchBar.setText(selected);
+                    suggestListView.setVisibility(View.INVISIBLE);
+                    smartSearch.search(selected, 1000);
```

Comment:
What does the 1000 mean?

Reply:
add comment into code 1000 means 1000ms(1s)